### PR TITLE
feat: Add modal style parameter

### PR DESCRIPTION
**Pull request by hicrystal**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [x] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Perfermance optimization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

https://github.com/ant-design/ant-design/issues/17991
This ticket will expose parameters to custom the modal component.
For header title :  `@modal-header-title-font-size`  and  `@modal-header-title-line-height` will help developers control the header title style.
For header border: ` @modal-header-border-width`  and `@modal-header-border-style` help developers control header border-bottom style.
For footer border: ` @modal-footer-border-width`  and `@modal-footer-border-style` help developers control footer border-top style.
For modal body padding: `@modal-header-padding-vertical` and ` @modal-header-padding-horizontal`  help developers control  body padding.

### 💡 Background and solution
The modal component is not flexible to overwrite the style theme.

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |      add parameters  for modal  component   |
| 🇨🇳 Chinese |       给Modal组件增加了更多的less变量    |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/style/themes/default.less [hunk 1]
```diff
@@ -484,18 +484,26 @@
...
+@modal-header-padding-vertical: @padding-md;
+@modal-header-padding-horizontal: @padding-lg;
 @modal-body-padding: @padding-lg;
 @modal-header-bg: @component-background;
-@modal-header-padding: @padding-md @padding-lg;
+@modal-header-padding: @modal-header-padding-vertical @modal-header-padding-horizontal;
+@modal-header-border-width: @border-width-base;
+@modal-header-border-style: @border-style-base;
+@modal-header-title-line-height: @font-size-lg;
+@modal-header-title-font-size: 22px;
```
Comment (yoyo837) [inline components/style/themes/default.less:495]:
```suggestion
@modal-header-title-font-size: @font-size-lg;
```

File: components/style/themes/default.less [hunk 2]
```diff
@@ -484,18 +484,26 @@
...
 // --
+@modal-header-padding-vertical: @padding-md;
+@modal-header-padding-horizontal: @padding-lg;
 @modal-body-padding: @padding-lg;
 @modal-header-bg: @component-background;
-@modal-header-padding: @padding-md @padding-lg;
+@modal-header-padding: @modal-header-padding-vertical @modal-header-padding-horizontal;
+@modal-header-border-width: @border-width-base;
+@modal-header-border-style: @border-style-base;
+@modal-header-title-line-height: @font-size-lg;
```
Comment (yoyo837) [inline components/style/themes/default.less:494]:
```suggestion
@modal-header-title-line-height: 22px;
```

File: components/style/themes/default.less [hunk 3]
```diff
@@ -484,18 +484,26 @@
 
 // Modal
 // --
+@modal-header-padding-vertical: @padding-md;
```
Comment (xrkffgg) [inline components/style/themes/default.less:487]: 注意下 themes/compact.less 紧凑模式，是否需要调整。

**Review by xrkffgg — COMMENTED** [review]
注意下

File: components/style/themes/default.less (hunk 1 above)
Comment (yoyo837) [reply components/style/themes/default.less:495]: 两个刚好弄反了，修改了一个，这里还有一个

Comment (hicrystal) [reply components/style/themes/default.less:495]: 不好意思 changed

File: components/style/themes/default.less (hunk 3 above)
Comment (hicrystal) [reply components/style/themes/default.less:487]: @xrkffgg `components/style/themes/compact.less` 里有改动

**Review by afc163 — APPROVED** [review]

**Pull request merged**